File: judge-api/src/main/kotlin/io/leonfoliveira/judge/api/dto/response/ProblemFullResponseDTO.kt
package io.leonfoliveira.judge.api.dto.response

import io.leonfoliveira.judge.core.domain.entity.Problem
import java.util.UUID

data class ProblemFullResponseDTO(
    val id: UUID,
    val letter: Char,
    val title: String,
    val description: AttachmentResponseDTO,
    val timeLimit: Int,
    val testCases: AttachmentResponseDTO,
)

fun Problem.toFullResponseDTO(): ProblemFullResponseDTO {
    return ProblemFullResponseDTO(
        id = id,
        letter = letter,
        title = title,
        description = description.toResponseDTO(),
        timeLimit = timeLimit,
        testCases = testCases.toResponseDTO(),
    )
}

File: applications/webapp/src/core/repository/dto/response/problem/ProblemFullResponseDTO.ts
import { Attachment } from "@/core/domain/model/Attachment";

export type ProblemFullResponseDTO = {
  id: string;
  letter: string;
  title: string;
  description: Attachment;
  timeLimit: number;
  memoryLimit: number;
  testCases: Attachment;
};
